src/api_types.rs

```diff
@@ -0,0 +1,107 @@
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct APIResponse {
+    pub date: DateResponse,
+    pub games: Vec<GameResponse>,
+    pub errors: Option<HashMap<String, serde_json::Value>>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct DateResponse {
+    pub raw: String,
+    pub pretty: String,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+#[allow(non_snake_case)]
```

Comment:
You can use snake_case variable names while reciving camelCase or PascalCase if instead of #[allow(non_snake_case)] you use #[serde(rename_all = "camelCase")]. I got this info from the [serde book](https://serde.rs/) if you interested.

Reply:
Thank you so much! That's exactly what I want to do but couldn't figure out how.